pkg/infrastructure/ibmcloud/clusterapi/clusterapi.go

```diff
@@ -369,11 +369,19 @@ func (p Provider) Ignition(ctx context.Context, in clusterapi.IgnitionInput) ([]
 	}
 	logrus.Debugf("bootstrap ignition config upload complete to %s/%s/%s", cosInstanceName, bucketName, ignitionFile)
 
-	ignitionURL := url.URL{
-		Scheme: "https",
-		Host:   cosEndpoint,
-		Path:   fmt.Sprintf("%s/%s", bucketName, ignitionFile),
+	// Build the URL for the ignition config.
+	cosURL, err := url.Parse(cosEndpoint)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse ibmcloud cos url: %s", cosURL)
+	}
+	// Make sure the COS URL has an https scheme, if one isn't already set.
+	if cosURL.Scheme == "" {
+		cosURL.Scheme = "https"
 	}
+	ignitionURL := cosURL.JoinPath(bucketName, ignitionFile)
+	//	Scheme: "https",
+	//	Host:   cosEndpoint,
+	//	Path:   fmt.Sprintf("%s/%s", bucketName, ignitionFile),
```

Comment:
nit: we can just remove these instead of commenting.

Reply:
yep, forgot to clean those up